test/integration/main_test.go

```diff
@@ -0,0 +1,416 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"testing"
+
+	"cloud.google.com/go/iam/apiv1/iampb"
+	"github.com/abcxyz/access-on-demand/pkg/cli"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/sethvargo/go-retry"
+	"google.golang.org/genproto/googleapis/type/expr"
+	"google.golang.org/protobuf/testing/protocmp"
+
+	resourcemanager "cloud.google.com/go/resourcemanager/apiv3"
+)
+
+var (
+	cfg           *config
+	projectClient *resourcemanager.ProjectsClient
+)
+
+func TestMain(m *testing.M) {
+	os.Exit(func() int {
+		ctx := context.Background()
+
+		if strings.ToLower(os.Getenv("TEST_INTEGRATION")) != "true" {
+			log.Printf("skipping (not integration)")
+			// Not integration test. Exit.
+			return 0
+		}
+
+		// set up global test config.
+		c, err := newTestConfig(ctx)
+		if err != nil {
+			log.Printf("Failed to parse integration test config: %v", err)
+			return 2
+		}
+		cfg = c
+
+		// Set up global resource manager client.
+		pc, err := resourcemanager.NewProjectsClient(ctx)
+		if err != nil {
+			log.Printf("failed to create projects client: %v", err)
+			return 2
+		}
+		defer pc.Close()
+		projectClient = pc
+
+		return m.Run()
+	}())
+}
+
+func TestIAMHandle(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name         string
+		filePath     string
+		wantBindings []*iampb.Binding
+		wantOutput   string
+	}{
+		{
+			name:     "success",
```

Comment:
Since we only have one scenario to test and likely we will keep it that way, I recommend get rid of the table test